test_result.go

```diff
@@ -81,10 +85,17 @@ func getCallerInfo(callers []uintptr) callerInfo {
 		logFn: f.Function,
 	}
 
-	// TODO: Skip t.Helper() frames.
-	f, _ = frames.Next()
-	if f == (runtime.Frame{}) {
-		return ci
+	skip := true
+	for skip {
+		// TODO: Skip t.Helper() frames.
+		f, _ = frames.Next()
+		if f == (runtime.Frame{}) {
+			return ci
+		}
+
+		// When a defer is triggered by a panic, it's added to the trace
+		// but panic is not shown as a log caller.
+		skip = f.Function == "runtime.gopanic"
```

Comment:
TIL that the callers of a `defer` vary depending on if the function returns or panics:
https://go.dev/play/p/sgaorZMqtxS

when there's no panic, we see:
```
defer w/o panic main.f1
defer w/o panic main.main
```

with a panic, we see:
```
defer after panic runtime.gopanic
defer after panic main.f2
defer after panic main.main
```

Looking at real `go test` logs, the caller is always the test code, not panic, so we skip here for consistency.